docs/reference/contrib/settings.rst

```diff
@@ -219,3 +222,47 @@ Or, alternately, using the ``set`` tag:
 .. code-block:: html+jinja
 
     {% set social_settings=settings("app_label.SocialMediaSettings") %}
+
+
+Utilising ``select_related`` to improve efficiency
+--------------------------------------------------
+
+For models with foreign key relationships to other objects (e.g. pages),
+which are very often needed to output values in templates, you can set
+the ``select_related`` attribute on your model class to have Wagtail
+fetch the related data for those fields when looking up setting for a site.
+
+Doing this results in a more complex initial query, but means that further
+additional queries will not be required to access the field values,
+meaning fewer database queries overall.
+
+Reusing the ``ImportantPages`` example from the previous section, here
+is how ``select_related`` can be used to improve efficiency:
+
+.. code-block:: python
+    :emphasize-lines: 4,5
+
+    @register_setting
+    class ImportantPages(BaseSetting):
+
+        # Fetch these pages when looking up ImportantPages for or a site
+        select_related = ["donate_page", "sign_up_page"]
+
+        donate_page = models.ForeignKey(
+            'wagtailcore.Page', null=True, on_delete=models.SET_NULL, related_name='+')
+        sign_up_page = models.ForeignKey(
+            'wagtailcore.Page', null=True, on_delete=models.SET_NULL, related_name='+')
+
+        panels = [
+            PageChooserPanel('donate_page'),
+            PageChooserPanel('sign_up_page'),
+        ]
+
+With these additions, the following template code would now only
```

Comment:
"would now only result in a single database query" -> "will now generate a single database query"?